lib/src/models/view/builders/timer_builder.dart

```diff
@@ -0,0 +1,31 @@
+import "package:rover_dashboard/models.dart";
+
+/// A [ValueBuilder] to modify a [MissionTimer]
+class TimerBuilder extends ValueBuilder<MissionTimer> {
+  /// Name of timer
+  String name = "";
+  
+  /// Number of minutes
+  NumberBuilder<int> duration = NumberBuilder<int>(0);
+
+  @override
+	bool get isValid => duration.isValid && name.isNotEmpty;
+
+  @override
+	MissionTimer get value => MissionTimer(
+    name: name,
+    duration: Duration(minutes: duration.value),
+  );
+
+  /// Updates Timer Name based on user input
+  void setName(String input) => name = input; 
+
+  /// Updates Timer duration based on input
+  void setTime(String input) => duration = NumberBuilder<int>(int.parse(input)); // Can always garuntee this is a int because of regex
```

Comment:
```suggestion
```
The idea of the `NumberBuilder` is that it should handle the validating and the `int.parse` for you. You should only need to hook up your UI to the `NumberBuilder` you already have instead of making a new one every time.

Since you didn't use `NumberEditor` (it can be really finicky), I made some small changes to make your `TextField` work with the number builder.